mne/io/snirf/_snirf.py

```diff
@@ -0,0 +1,193 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#
+# License: BSD (3-clause)
+
+import re as re
+import numpy as np
+import h5py
```

Comment:
@larsoner is it ok to use h5py rather than `from ...externals.pymatreader import read_mat`?

If I use `read_mat` then I get the following error which appears to be caused as the upstream software does not save a mat file type version.

```python

/Users/rluke/Library/Python/3.7/lib/python/site-packages/scipy/io/matlab/mio.py:72: in mat_reader_factory
    mjv, mnv = get_matfile_version(byte_stream)
...


    def get_matfile_version(fileobj):
        """
        Return major, minor tuple depending on apparent mat file type
    
        Where:
    ...


        if maj_val in (1, 2):
            return ret
>       raise ValueError('Unknown mat file type, version %s, %s' % ret)
E       ValueError: Unknown mat file type, version 0, 0

```

Reply:
It's worth opening a pymatreader issue first. Maybe for example you can override/specify the version manually. It'll probably be easier than using h5py manually.

Also I would check to see how MATLAB handles this. pymatreader should probably do what MATLAB does, if reasonable.

I'd also bother the vendor to fix their files. It's probably 1 line of code for them to change...

But yes you can use h5py directly if you need to.

Reply:
> I'd also bother the vendor to fix their files. It's probably 1 line of code for them to change...

Yep tried already. They aren't going to change it.

> But yes you can use h5py directly if you need to.

Ok I will try your suggestions above, modify pymatreader and see what Matlab does. But keep this as a final option.